packages/abi/src/matchers/sway-type-matchers.ts

```diff
@@ -0,0 +1,89 @@
+type SwayType =
+  | 'empty'
+  | 'bool'
+  | 'u8'
+  | 'u16'
+  | 'u32'
+  | 'u64'
+  | 'u256'
+  | 'b256'
+  | 'generic'
+  | 'string'
+  | 'stdString'
+  | 'option'
+  | 'result'
+  | 'enum'
+  | 'struct'
+  | 'b512'
+  | 'bytes'
+  | 'vector'
+  | 'tuple'
+  | 'array'
+  | 'assetId'
+  | 'evmAddress'
+  | 'rawUntypedPtr' // might not need it
+  | 'rawUntypedSlice'; // might not need it
```

Comment:
I question the need for these last two types and we'll review if we need them later on when we start implementing more.

Reply:
@nedsalk Why they might not be needed?

Reply:
We don't use them for encoding and decoding, nor for generating types in typegen.

Reply:
@nedsalk Thanks for the clarification